### PR TITLE
feat(odyssey-react): add Box as root of FieldGroup

### PR DESCRIPTION
**Pull request by andrewberg-okta**

### CODE REVIEW

**Review by edburyenegren-okta — APPROVED** [review]
No crossover declarations in FieldGroup's module -- good to go!

File: packages/odyssey-react/src/components/FieldGroup/FieldGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+};
+
+let FieldGroup: FunctionComponent<Props> & Statics = Object.assign(
+  (props: Props) => {
+    const { children, title, desc, ...rest } = props;
+
+    const omitProps = useOmit(rest);
+
+    return (
+      <Box as="fieldset" {...omitProps} className={styles.root}>
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/FieldGroup/FieldGroup.tsx:57]: sanity checking [if you remove the global styles for typography and reset in storybook here ](https://github.com/okta/odyssey/blob/develop/packages/odyssey-storybook/.storybook/preview.js#L1) does FieldGroup now render just fine? At this point we should be OK to perform these types of checks now that Box and Text have the previously global styles encapsulated

File: packages/odyssey-react/src/components/FieldGroup/index.tsx [hunk 2]
```diff
@@ -10,71 +10,4 @@
...
-FieldGroup.displayName = "FieldGroup";
-Error.displayName = "FieldGroupError";
-
-FieldGroup.Error = Error;
-
-FieldGroup = withStyles(styles)(FieldGroup);
-
-export type FieldGroupProps = ComponentProps<typeof FieldGroup>;
-export { FieldGroup };
+export * from "./FieldGroup";
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/FieldGroup/index.tsx:13]: thank you for creating a barrel file. It does not need a `.tsx` extension however you can use a plain `.ts` here

File: packages/odyssey-react/src/components/FieldGroup/FieldGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+interface PropsError {
+  children: ReactNode;
+}
+
+type Statics = {
+  Error: typeof Error;
+};
+
+let FieldGroup: FunctionComponent<Props> & Statics = Object.assign(
```
Comment (jeffbelser-okta) [inline packages/odyssey-react/src/components/FieldGroup/FieldGroup.tsx:50]: Trying to recall why we didn't forward the ref here...

**Review by jeffbelser-okta — APPROVED** [review]
strong work 💪 please double check the rendering works fine without the global styles in storybook

**Pull request merged**